spec/models/event_spec.rb

```diff
@@ -1,8 +1,152 @@
 require 'spec_helper'
 
 describe Event do
+  subject { create(:event) }
+  let(:conference) { create(:conference) }
+  let(:event) { create(:event, program: conference.program) }
+  let(:new_event) { create(:event) }
+  let(:user) { create(:user) }
 
-  describe 'abstract_word_count' do
+  describe 'association' do
+    it 'should belong to program' do
+      is_expected.to belong_to :program
+    end
+
+    it 'should belong to event_type' do
+      is_expected.to belong_to :event_type
+    end
+  end
+
+  describe 'validation' do
+    it 'has a valid factory' do
+      expect(build(:event)).to be_valid
+    end
+
+    it { is_expected.to validate_presence_of(:title) }
+    it { is_expected.to validate_presence_of(:abstract) }
+    it { is_expected.to validate_presence_of(:program) }
+    it { is_expected.to validate_presence_of(:event_type) }
+
+    describe '#abstract_limit' do
+      before :each do
+        event.event_type.maximum_abstract_length = 2
+        event.event_type.minimum_abstract_length = 2
+      end
+
+      context 'is invalid' do
+        it 'when abstract is too long' do
+          event.abstract = 'Test abstract here'
+          expect(event.valid?).to eq false
+          expect(event.errors[:abstract]).to eq ['cannot have more than 2 words']
+        end
+
+        it 'when abstract is too short' do
+          event.abstract = 'Test'
+          expect(event.valid?).to eq false
+          expect(event.errors[:abstract]).to eq ['cannot have less than 2 words']
+        end
+      end
+
+      context 'is valid' do
+        it 'when abstract length is within limits' do
+          event.abstract = 'Test abstract'
+          expect(event.valid?).to eq true
+          expect(event.errors.size).to eq 0
+        end
+      end
+    end
+
+    describe '#before_end_of_conference' do
+      context 'is invalid' do
+        it 'when event is created after the conference end_date, and returns an error message' do
+          conference = create(:conference, start_date: Date.today - 1, end_date: Date.today - 1)
+          new_event = build(:event, program: conference.program)
+          expect(new_event.valid?).to eq false
+          expect(new_event.errors[:created_at]).to eq ["can't be after the conference end date!"]
+        end
+      end
+
+      context 'is valid' do
+        it 'when event is created before the conference end_date' do
+          conference = create(:conference, start_date: Date.today - 1, end_date: Date.today + 1)
+          new_event = build(:event, program: conference.program)
+          expect(new_event.valid?).to eq true
+        end
+      end
+    end
+  end
+
+  describe 'scope ' do
+    context 'confirmed' do
+      before :each do
+        conference = create(:conference)
+        create(:event, program: conference.program)
+      end
+
+      it 'returns only confirmed events' do
+        my_event = create(:event, state: 'confirmed', program: conference.program)
```

Comment:
and when you use conference here, you use the conference create in the let in line 5 and not the conference created in the before block! So either remove line 82 if you don't need it or move it into a let or use an instance variable.